app/src/display/widgets/hid_indicators_status.c

```diff
@@ -0,0 +1,69 @@
+
+/*
+ * 
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display.h>
+#include <zmk/display/widgets/hid_indicators_status.h>
+#include <zmk/hid_indicators.h>
+#include <zmk/events/hid_indicators_changed.h>
+#include <zmk/hid_indicators_types.h>
+
+#define ZMK_LED_NUMLOCK_BIT BIT(0)
+#define ZMK_LED_CAPSLOCK_BIT BIT(1)
+#define ZMK_LED_SCROLLLOCK_BIT BIT(2)
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+
+struct hid_indicators_status_state {
+    zmk_hid_indicators_t flags;  // HID Indicator Status Bit Mask
+};
+
+static void set_hid_indicators_symbol(lv_obj_t *label, struct hid_indicators_status_state state) {
+    char text[10];
+    snprintf(text, sizeof(text), "%s%s%s",
+             (state.flags & ZMK_LED_CAPSLOCK_BIT) ? "C " : "",
+             (state.flags & ZMK_LED_NUMLOCK_BIT) ? "N " : "",
+             (state.flags & ZMK_LED_SCROLLLOCK_BIT) ? "S " : "");
+    if (text[0] == '\0') {
+        strcpy(text, "---");
```

Comment:
I don't like the 3 dashes, also they do not equal the width of the letters above, because of the missing spaces.
Suggestion, one could use the "" in the else path above to indicate an 'of' state. e.g.: `(state.flags & ZMK_LED_CAPSLOCK_BIT) ? "C " : "- ",` or `: "--",`